src/cryptonote_core/service_node_swarm.cpp

```diff
@@ -97,16 +95,16 @@ namespace service_nodes
     if (has_starving_swarms)
       return;
 
+    std::vector<excess_pool_snode> pool_snodes;
+    std::vector<crypto::public_key> new_swarm_snodes;
```

Comment:
Although you move out of `new_swarm_snodes`, I don't believe it is guaranteed to be emptied (This definitely falls under *premature optimisation* as we potentially introduced a bug without any measurable performance gain). I would put it back inside the loop or at least `clear` it explicitly.

Reply:
Mmh the literature on std::move isn't super clear. Some of sites say that the moved object is left in a default state, but here it says unspecified state:
`Although note that -in the standard library- moving implies that the moved-from object is left in a valid but unspecified state. Which means that, after such an operation, the value of the moved-from object should only be destroyed or assigned a new value; accessing it otherwise yields an unspecified value.`
Regardless, I should `clear` it !

Reply:
Best move it back inside the loop actually.